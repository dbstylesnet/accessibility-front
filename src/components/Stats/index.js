import React from 'react'
import PropTypes from 'prop-types'
import StatsDetails from './styles'
import StatsBox from '../StatsBox'

export const Stats = ({
    salesOverview = {
        uploads: 20,
        successfulUploads: 10,
        linesAttempted: 20,
        linesSaved: 15,
    }
}) => {
    const { uploads, successfulUploads, linesAttempted, linesSaved } = salesOverview
    const uploadCalculated =  successfulUploads / uploads * 100
    const linesCalculated = linesSaved / linesAttempted * 100
    
    return <div 
        data-testid='statsContainer'
        aria-label="Uploads and lines saved - statistics"> 
        <StatsDetails>
            <StatsBox 
                calculated={uploadCalculated} 
                description={'Upload Success'}
                data-testid='uploadsSuccessRate'
                />
            <StatsBox 
                calculated={linesCalculated} 
                description={'Lines Saved'}
                data-testid='linesSuccessRate' 
                />
        </StatsDetails>
    </div>
}

export default Stats

StatsBox.propTypes = {
    calculated: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
}
