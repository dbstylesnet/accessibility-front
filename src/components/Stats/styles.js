import styled from 'styled-components'

const StatsDetails = styled.div`
  display: flex;
  div:first-child {
    margin-right: 1px;
  }
  div {
    font-size: 1em;
  }
  @media only screen and (max-width: 800px) {
    flex-direction: column;
    div:first-child {
      margin-right: 0px;
      margin-bottom: 1px;
    }
  }
`

export default StatsDetails
