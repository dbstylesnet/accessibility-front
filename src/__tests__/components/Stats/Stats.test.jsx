import { render, screen } from '@testing-library/react';
import Stats from '../../../components/Stats';
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom'

test('renders Stats component', () => {
    render(<Stats />);
    const elements = screen.queryAllByText(/%/i);
    expect(elements).toBeTruthy();

});


test('renders Stats component correctly', () => {
    const tree = TestRenderer
      .create(<Stats />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
