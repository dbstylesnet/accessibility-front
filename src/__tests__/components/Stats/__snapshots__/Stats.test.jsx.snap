// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Stats component correctly 1`] = `
<div
  aria-label="Uploads and lines saved - statistics"
  data-testid="statsContainer"
>
  <div
    className="sc-bdfBwQ fDqpIs"
  >
    <div
      className="sc-gsTCUz leBjBn"
    >
      <div>
        50
         % 
      </div>
      <div>
        Upload Success
      </div>
    </div>
    <div
      className="sc-gsTCUz leBjBn"
    >
      <div>
        75
         % 
      </div>
      <div>
        Lines Saved
      </div>
    </div>
  </div>
</div>
`;
